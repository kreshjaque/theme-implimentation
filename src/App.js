import React from "react";
import {ThemeProvider} from "styled-components";
import { GlobalStyles } from "./components/GlobalStyles";
import { lightTheme, darkTheme } from "./components/Themes"
import "./App.css";
import Toggle from "./components/Toggler";
import {useDarkMode} from "./components/useDarkMode";
import ThemeSwitch from "./components/ThemeSwitch";

const App = () => {
    const {theme, themeToggler, themeTogglerWithParam} = useDarkMode();
    const themeMode = theme === 'light' ? lightTheme : darkTheme;
  return (
      <ThemeProvider theme={themeMode}>
        <>
          <GlobalStyles/>
          <div className="App">
              <Toggle theme={theme} toggleTheme={themeToggler} />
              <h1>Alive</h1>
              <ThemeSwitch theme={theme} switchTheme={themeTogglerWithParam} />
            <p>Still alive</p>
          </div>
        </>
      </ThemeProvider>

  );
};
export default App;
