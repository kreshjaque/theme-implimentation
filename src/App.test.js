import {render} from '@testing-library/react';
import App from './App';

test('renders the app', () => {
    const content = render(<App/>);
    expect(content).toMatchSnapshot();
});
