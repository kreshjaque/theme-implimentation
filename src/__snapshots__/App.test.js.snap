// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <button
          class="sc-bdfBwQ eiPQmm"
        >
          Switch Theme
        </button>
        <h1>
          Alive
        </h1>
        <input
          checked=""
          class="sc-gsTCUz drHVHv"
          id="light"
          name="theme"
          type="radio"
          value="light"
        />
        <label
          for="light"
        >
          Light
        </label>
        <br />
        <input
          class="sc-gsTCUz drHVHv"
          id="dark"
          name="theme"
          type="radio"
          value="dark"
        />
        <label
          for="dark"
        >
          Dark
        </label>
        <br />
        <p>
          Still alive
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <button
        class="sc-bdfBwQ eiPQmm"
      >
        Switch Theme
      </button>
      <h1>
        Alive
      </h1>
      <input
        checked=""
        class="sc-gsTCUz drHVHv"
        id="light"
        name="theme"
        type="radio"
        value="light"
      />
      <label
        for="light"
      >
        Light
      </label>
      <br />
      <input
        class="sc-gsTCUz drHVHv"
        id="dark"
        name="theme"
        type="radio"
        value="dark"
      />
      <label
        for="dark"
      >
        Dark
      </label>
      <br />
      <p>
        Still alive
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
